virtcontainers/clh.go

```diff
@@ -421,28 +378,58 @@ func (clh *cloudHypervisor) addDevice(devInfo interface{}, devType deviceType) e
 	defer span.Finish()
 
 	var err error
+	devLogger := clh.Logger().WithFields(log.Fields{
+		"function": "addDevice",
+	})
 
 	switch v := devInfo.(type) {
 	case Endpoint:
-		clh.Logger().WithField("function", "addDevice").Infof("Adding Endpoint of type %v", v)
-		clh.cliBuilder.AddNet(CLINet{
-			device: v.Name(),
-			mac:    v.HardwareAddr(),
-		})
+		devLogger.Infof("Adding Endpoint of type %v", v)
+		n := chclient.NetConfig{}
+		n.Mac = v.HardwareAddr()
+		n.Tap = v.NetworkPair().TapInterface.TAPIface.Name
+
+		//TODO server does not allow empty IP fail serde to get it
+		n.Ip = "0.0.0.0"
+		n.Mask = "0.0.0.0"
+		clh.vmconfig.Net = append(clh.vmconfig.Net, n)
+
 	case types.HybridVSock:
-		clh.Logger().WithFields(log.Fields{
-			"function": "addDevice",
-			"path":     v.UdsPath,
-			"cid":      v.ContextID,
-			"port":     v.Port,
+		devLogger.WithFields(log.Fields{
+			"path": v.UdsPath,
+			"cid":  v.ContextID,
+			"port": v.Port,
 		}).Info("Adding HybridVSock")
-		clh.cliBuilder.SetVsock(&CLIVsock{
-			cid:        uint32(v.ContextID),
-			socketPath: v.UdsPath,
-			iommu:      false,
-		})
+		clh.vmconfig.Vsock = []chclient.VsockConfig{{Cid: defaultGuestVSockCID, Sock: v.UdsPath}}
+	case types.Volume:
+
+		if clh.config.SharedFS != config.VirtioFS {
```

Comment:
does it support block based volumes?

Reply:
I have not tried but is not possible to hotAdd volumes,  for K8s this the only way to go.